Views/Lombiq.Privacy.ConsentBanner.cshtml

```diff
@@ -1,67 +1,87 @@
-    Script.Require(ResourceNames.JsConsentService).AtHead();
+    var bannerId = "privacy-consent-banner";
+    var acceptButtonId = "privacy-consent-accept-button";
+    var deniedButtonId = "privacy-consent-denied-button";
 
-    // Setting the authenticated flag for authenticated users and not executing the rest of the template because it's
-    // not needed.
-    if (WorkContext.CurrentUser != null)
-    {
-        Script.Include("authenticated-flag-setter.js").AtHead();
-        return;
-    }
-
-    Script.Require("jQueryUI");
-    Style.Require("jQueryUI");
+    var consentFeature = Context.Features.Get<ITrackingConsentFeature>();
+    var acceptCookie = consentFeature?.CreateConsentCookie();
+    var deniedCookie = acceptCookie.Replace("yes", "no");
 
-    Style.Require(ResourceNames.ConsentBanner);
+    var isRegistrationConsentFeatureEnabled = (await shellFeaturesManager.GetEnabledFeaturesAsync())
+        .Any(feature => feature.Id == FeatureNames.RegistrationConsent);
 
-    const string ConsentBannerId = "consent-banner";
+    // This represents the case when a user registered before the Registration consent feature was enabled
+    // so they need to accept the consent now.
+    var isAuthenticatedHasNotConsent = isRegistrationConsentFeatureEnabled && User.Identity.IsAuthenticated;
 }
 
+<shape Type="Lombiq_Privacy_ConsentBanner_Resources" />
 
-@Display.Lombiq_Privacy_ConsentBanner_Resources()
-
-<div id="@ConsentBannerId" class="consentBanner" title="@T("Privacy consent")" hidden>
-    <span>
-        @Display.Lombiq_Privacy_ConsentBanner_Text()
-    </span>
+<div id="@bannerId" class="privacy-consent-banner">
+    <div class="modal-content">
+        <div class="modal-header bg-secondary">
+            <h5 class="modal-title" id="exampleModalLabel">@T["Privacy consent"]</h5>
+        </div>
+        <div class="modal-body">
+            <shape Type="Lombiq_Privacy_ConsentBanner_Text" />
+        </div>
+        <div class="modal-footer justify-content-center">
+            <div class="btn-group btn-group-lg" role="group">
+                <button id="@acceptButtonId" type="button" class="btn btn-success" data-cookie-string="@acceptCookie">
+                    @T["I agree, do not show me this any more"]
+                </button>
+                @if (!isAuthenticatedHasNotConsent)
+                {
+                    <button id="@deniedButtonId" type="button" class="btn btn-secondary" data-cookie-string="@deniedCookie">
+                        @T["I don't agree, do not show me this any more"]
+                    </button>
+                }
+            </div>
+        </div>
+    </div>
 </div>
 
+@if (!isAuthenticatedHasNotConsent)
+{
+    <script depends-on="jQuery" at="Foot">
+        $("#@acceptButtonId").click(function () {
+            document.cookie = $("#@acceptButtonId")[0].dataset.cookieString;
+            $("#@bannerId").remove();
+        });

```

Comment:
Ilyeneket is a jQuery document.ready-jében kellene, lásd: https://orcharddojo.net/orchard-resources/CoreLibrary/DevelopmentGuidelines/BestPractices/JavaScript